docs/source/reference/indexing.rst

```diff
@@ -28,6 +28,7 @@ Properties
    Index.inferred_type
    Index.is_all_dates
    Index.shape
+   Index.map
```

Comment:
Shall we also add `CategoricalIndex.map`?

Following https://pandas.pydata.org/docs/reference/indexing.html#id1, we'd better add "Modifying and computations" section under "CategoricalIndex".